torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -111,29 +116,117 @@ def create_subclass_meta(
-        xs_inner = []
-        for x in xs:
-            if is_traceable_wrapper_subclass(x):
-                xs_inner.extend(get_plain_tensors(typing.cast(Tensor, x)))
-            else:
-                xs_inner.append(x)
-        return xs_inner
 
+
+def filter_symints(size: Iterable[Union[int, SymInt]]) -> List[SymInt]:
+    return [s for s in size if isinstance(s, SymInt)]
+
+
+# The reason for "append_symints"
+#
+# * At compile time: we append extra symint args when unwrapping primals
+# (but not tangents, because they should always share symints with primals).
+# We also append extra symints when unwrapping the subclass outputs of the
+# traced function, so we can return them as extra outputs
+#
+# * At runtime: we similarly append subclass sizes when we unwrap subclass
+# primals (but not tangents) on entry to the forward.
+def unwrap_tensor_subclasses(
+    wrapped_args: List[Union[Tensor, int]],
+    *,
+    is_runtime: bool,
+    append_symints: bool,
+    subclass_metas: Optional[List[Union[int, SubclassCreationMeta]]] = None,
+):
+    xs_inner: List[Union[int, Tensor, SymInt]] = []
+
+    for idx, x in enumerate(wrapped_args):
+        if is_traceable_wrapper_subclass(x):
+            size = x.size()
+            stride = x.stride()
+            xs_inner.extend(get_plain_tensors(typing.cast(Tensor, x)))
+            if append_symints:
+                if is_runtime:
+                    assert subclass_metas is not None
+                    meta = subclass_metas[idx]
+                    assert isinstance(meta, SubclassCreationMeta)
+
+                    # outer_size
+                    symint_placeholders = [type(i) is not int for i in meta.outer_size]
+                    assert len(size) == len(symint_placeholders)
+                    xs_inner.extend(
+                        [
+                            r
+                            for (r, is_symint) in zip(size, symint_placeholders)
+                            if is_symint
+                        ]
+                    )
+
+                    # outer_stride
+                    symint_placeholders = [
+                        type(i) is not int for i in meta.outer_stride
+                    ]
+                    assert len(stride) == len(symint_placeholders)
+                    xs_inner.extend(
+                        [
+                            r
+                            for (r, is_symint) in zip(stride, symint_placeholders)
+                            if is_symint
+                        ]
+                    )
+                else:
+                    xs_inner.extend(filter_symints(size))
+                    xs_inner.extend(filter_symints(stride))
+        else:
+            xs_inner.append(x)
+    return xs_inner
+
+
+def unwrap_tensor_subclasses_joint(
+    wrapped_args,
+    *,
+    is_runtime: bool,
+    append_symints: bool,
+    subclass_metas: Optional[List[Union[int, SubclassCreationMeta]]],
```

Comment:
Per the comments [here](https://github.com/pytorch/pytorch/pull/125941/files#r1717189366), I think:

(1) it didn't really make sense to m that you can pass in a single value for `subclass_metas`, that you can re-use for both the primals and tangents (since in they can have different subclass metadat)

(2) in fact, it looks like the only place you were calling this function with subclass_metas plumbed in was in the above linked comment, where you don't actually want to use this joint function.

 So I think we should either kill this `subclass_metas` argument, or kill `unwrap_tensor_subclasses_joint` entirely (and just call the underlying `unwrap_tensor_subclasses()` when necessary)

Reply:
Removed `unwrap_tensor_subclasses_joint`